### ISSUE TITLE
[Bug]: Wrong result when using the leading column of a unique index.

### ISSUE THREAD

**Issue by fengttt** [matrixorigin/matrixone#14506]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch Name

main

### Commit ID

060f904

### Other Environment Information

```Markdown
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

Wrong result, see repro

Create table with unique index, then populate the table with enough data to trigger query plan using BlockFilterCond.    Result is wrong.

### Steps to Reproduce

```Markdown
mysql> create table t(i int, j int, k int, unique key (i, j));
Query OK, 0 rows affected (0.01 sec)

mysql> insert into t select result, result, result from generate_series(1, 100) tmpt;
Query OK, 100 rows affected (0.02 sec)

mysql> insert into t select i, null, null from t;
Query OK, 100 rows affected (0.02 sec)

mysql> explain select * from t where i = 2;
+--------------------------------+
| QUERY PLAN                     |
+--------------------------------+
| Project                        |
|   ->  Table Scan on repro.t    |
|         Filter Cond: (t.i = 2) |
+--------------------------------+
3 rows in set (0.00 sec)

mysql> insert into t select i, null, null from t;
^[[AQuery OK, 204800 rows affected (0.19 sec)

mysql> explain select * from t where i < 2;
+--------------------------------------+
| QUERY PLAN                           |
+--------------------------------------+
| Project                              |
|   ->  Table Scan on repro.t          |
|         Filter Cond: (t.i < 2)       |
|         Block Filter Cond: (t.i < 2) |
+--------------------------------------+

mysql> select * from t where i = 42;
+------+------+------+
| i    | j    | k    |
+------+------+------+
|   42 |   42 |   42 |
+------+------+------+
1 row in set (0.02 sec)
```

**Comment by fengttt** [comment]
mysql> explain analyze select * from t where i = 42;
+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| QUERY PLAN                                                                                                                                                                           |
+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Project                                                                                                                                                                              |
|   Analyze: timeConsumed=0ms waitTime=4ms inputRows=1 outputRows=1 InputSize=12bytes OutputSize=12bytes MemorySize=12bytes                                                            |
|   ->  Join                                                                                                                                                                           |
|         Analyze: timeConsumed=0ms probe_time=[0ms,0ms,0ms] build_time=[0ms] waitTime=20ms inputRows=8193 outputRows=1 InputSize=163848bytes OutputSize=12bytes MemorySize=65772bytes |
|         Join Type: SEMI                                                                                                                                                              |
|         Join Cond: (t.__mo_fake_pk_col = __mo_index_unique_018d6599-d90f-7631-8788-79a8ad631f06.__mo_index_pri_col)                                                                  |
|         Runtime Filter Build: #[-1,0]                                                                                                                                                |
|         ->  Table Scan on repro.t                                                                                                                                                    |
|               Analyze: timeConsumed=1ms waitTime=0ms inputRows=8192 outputRows=8192 InputSize=163840bytes OutputSize=163840bytes MemorySize=327680bytes                              |
|               Block Filter Cond: (t.i = 42)                                                                                                                                          |
|               Runtime Filter Probe: t.__mo_fake_pk_col                                                                                                                               |
|         ->  Table Scan on repro.__mo_index_unique_018d6599-d90f-7631-8788-79a8ad631f06                                                                                               |
|               Analyze: timeConsumed=1ms waitTime=0ms inputRows=1 outputRows=1 InputSize=32bytes OutputSize=8bytes MemorySize=41bytes                                                 |
|               Filter Cond: prefix_eq(__mo_index_unique_018d6599-d90f-7631-8788-79a8ad631f06.__mo_index_idx_col, ':*')                                                               |
|               Block Filter Cond: prefix_eq(__mo_index_unique_018d6599-d90f-7631-8788-79a8ad631f06.__mo_index_idx_col, ':*')                                                         |
+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
15 rows in set (0.00 sec)

**Comment by aunjgr** [comment]
It has been fixed by #14498.

**Comment by fengttt** [comment]
@aunjgr is this the right/adequate fix?   I thought the problem was due to that values with null was not added to uniq index, so the "order" was not the root cause.
